e2e/specs/st_multiselect.spec.js

```diff
@@ -212,37 +187,109 @@ describe("st.multiselect", () => {
                 "value 7: []" +
                 "value 8: []" +
                 "value 9: []" +
+                "value 10: []" +
+                "value 11: []" +
                 "multiselect changed: False"
             );
         });
       });
     });
 
     it("calls callback if one is registered", () => {
-      cy.get(".stMultiSelect")
-        .should("have.length", 9)
-        .last()
+      cy.getIndexed(".stMultiSelect", 10)
         .find("input")
         .click();
       cy.get("li")
         .first()
         .click();
 
       cy.get("[data-testid='stText']")
-        .should("have.length", 10)
         .should(
-          "have.text",
-          "value 1: []" +
-            "value 2: ['female']" +
-            "value 3: []" +
-            "value 4: ['tea', 'water']" +
-            "value 5: []" +
-            "value 6: []" +
-            "value 7: []" +
-            "value 8: []" +
-            "value 9: ['male']" +
+          "contain.text",
+            "value 11: ['male']" +
             "multiselect changed: True"
         );
     });
   });
+
+  describe("when using max_selections for st.multiselect", () => {
+    beforeEach(() => {
+      // this is the 'close button' element for 'Male'
+      cy.get('.stMultiSelect span[data-baseweb="tag"] span:last-child')
```

Comment:
Two things here:
* Won't this affect all multiselects on the page rather than just the one that's actually being used in the test? This should be fixed to be more specific
* I think it'd be more readable to figure out a way to select the right tag to deselect based on the text in it if possible, that way you don't need to clarify what's going on with a comment.